csharp/test/Microsoft.ML.OnnxRuntime.Tests.Common/InferenceTest.cs

```diff
@@ -233,6 +248,32 @@ public void TestUpdatingEnvWithCustomLogLevel()
             ortEnvInstance.EnvLogLevel = LogLevel.Verbose;
             Assert.Equal(LogLevel.Verbose, ortEnvInstance.EnvLogLevel);
         }
+        
+        [Fact(DisplayName = "TestUpdatingEnvWithThreadingOptions")]
+        public void TestUpdatingEnvWithThreadingOptions()
+        {
+            using (var opt = new OrtThreadingOptions()) {
+                // Make sure we start anew
```

Comment:
is destroying and re-creating an OrtEnv now supported? I recall at one point it was not.

https://github.com/microsoft/onnxruntime/pull/10200#discussion_r780604498

Reply:
Well, it was always supported on the C# level and it is a part of the public interface for better or worse. On the C++ level it is a ref-counted object which for test purposes seems to perform as expected.

Reply:
Yes, we have to support repeated invocations and follow the documented pattern of IDispose no matter how strange it may seem.

Reply:
I suppose my question is - if `TestUpdatingEnvWithThreadingOptions` and `TestUpdateEnvMultipleTimesExpectException` both run in the same process, and they both call `OrtEnv.Instance().Dispose()`, does the second call actually do anything?

a related question is what's the rationale for this comment?
https://github.com/microsoft/onnxruntime/blob/5b2d43cff8d186018f7aed96bd7c43c7fa0636e8/csharp/src/Microsoft.ML.OnnxRuntime/OnnxRuntime.shared.cs#L285

Reply:
The second call would not do anything. And that is the required behavior.